packages/flutter/lib/src/widgets/preferred_size.dart

```diff
@@ -34,6 +33,81 @@ abstract class PreferredSizeWidget implements Widget {
 /// affect the child's layout in any way. It just advertises a preferred size
 /// which can be used by the parent.
 ///
+/// Widgets like [AppBar] implements a [PreferredSizeWidget], meaning the
+/// possibility of changing [Scaffold.appBar] behavior.
+///
+/// {@tool dartpad --template=stateless_widget_material}
+///
+/// This sample shows a custom AppBar with a [PreferredSize].
+/// The preferred size is set to 80 logical pixels.
+///
+/// Consider changing the [PreferredSize] value and see how the
+/// custom [AppBar] behaves. The benefit of using
+/// PreferredSize as a [Scaffold.appBar] child it's the possibility of creating
+/// custom and adapting appBars.
```

Comment:
It's probably better to stick to being descriptive rather than prescriptive here, not all users will want to use this API to create create an adapting AppBar. Instead you could simply say that the value changes the size:

```suggestion
/// This sample shows a custom AppBar with a [PreferredSize], with its height
/// set to 80 logical pixels. Changing the [PreferredSize] can be used to change
/// the height of the AppBar.
```